src/main/java/nextstep/subway/line/domain/Sections.java

```diff
@@ -65,4 +70,37 @@ public void validate(Section section) {
     public boolean contains(Station station) {
         return sections.stream().anyMatch(s -> s.containStation(station));
     }
+
+    public void delete(Station station) {
+        if (canNotDelete()) {
+            throw new CanNotDeleteStateException();
+        }
+        if (!contains(station)) {
+            throw new NotFoundException();
+        }
+
+        Section upSection = sections.stream().filter(s -> s.getDownStation() == station)
+                                             .findFirst().map(this::remove).orElse(null);
+        Section downSection = sections.stream().filter(s -> s.getUpStation() == station)
+                                               .findFirst().map(this::remove).orElse(null);
+
+        if (existSections(upSection, downSection)) {
+            Section section = new Section(upSection.getUpStation(), downSection.getDownStation(),
+                upSection.getDistance() + downSection.getDistance());
+            sections.add(section);
+        }
```

Comment:
```suggestion
        if (!existSections(upSection, downSection)) return;
        sections.add(
            new Section(
                upSection.getUpStation(),
                downSection.getDownStation(),
                upSection.getDistance() + downSection.getDistance()
            )
        );
```
if 안에서 무언가 해주기보단, if 밖에서 해주는게 어떨까요?